AGENTS.md

```diff
@@ -1,400 +1,311 @@
-./ci/test_cuda_cccl_examples_python.sh -py-version 3.10
-./ci/test_cuda_cccl_cooperative_python.sh -py-version 3.10
-
-# Using CMake presets - NEVER CANCEL, takes up to 15 minutes
 ctest --preset=cub-cpp17
 ```
 
-**Test configuration options:**
-- `-no-lid` - Run without limited device identifiers
-- `-lid0`, `-lid1`, `-lid2` - Test with specific device limitations
-- `-compute-sanitizer-memcheck` - Run with memory checking (slower)
+Options:
 
-## Python CCCL Packages
+* `-compute-sanitizer-memcheck` — Run with memory checking or other compute-sanitizer tools (not all projects support this)
 
-**CRITICAL**: Python packages require different build parameters than C++ components. Use `-py-version X.Y` format instead of `-cxx` and `-std` parameters.
+---
 
-**Available Python versions:** `3.10`, `3.11`, `3.12`, `3.13`
+## Python CCCL Packages
 
-### Python Package Components
+Python components require different parameters than C++ builds. Use `-py-version` instead of compiler flags.
 
-CCCL provides comprehensive Python bindings and algorithms:
+Supported versions: `3.10`, `3.11`, `3.12`, `3.13`
 
-**cuda.cccl.parallel** - Device-level parallel algorithms (experimental):
-- **Algorithms**: `reduce_into`, `exclusive_scan`, `inclusive_scan`, `radix_sort`, `merge_sort`, `histogram_even`, `unique_by_key`, `binary_transform`, `unary_transform`, `segmented_reduce`
-- **Iterators**: `CountingIterator`, `ConstantIterator`, `TransformIterator`, `ReverseInputIterator`, `ZipIterator`, `CacheModifiedInputIterator`
-- **Custom types**: `gpu_struct` decorator for user-defined data types
-- **Performance**: Hand-optimized CUDA kernels, portable across GPU architectures
+### Modules
 
-**cuda.cccl.cooperative** - Block and warp-level cooperative algorithms:
-- **Block-level primitives**: reduce, scan, sort operations within thread blocks
-- **Warp-level primitives**: cooperative operations within warps
-- **Integration**: Designed for use within Numba CUDA kernels
-- **Custom kernels**: Building blocks for advanced CUDA kernel development
+* **cuda.cccl.parallel** — Device-level algorithms, iterators, custom GPU types
+* **cuda.cccl.cooperative** — Block/warp-level primitives
+* **cuda.cccl.headers** — Programmatic access to headers
 
-**cuda.cccl.headers** - Include paths for CCCL headers in Python:
-- **Header access**: Programmatic access to libcudacxx, CUB, Thrust, and cudax headers
-- **Integration**: Use with Numba, PyCUDA, or other Python CUDA libraries
-- **Path management**: `get_include_paths()` function for compiler integration
+### Installation
 
-### Python Installation and Usage
+From PyPI:
 
-**Install from PyPI** (recommended for users):
 ```bash
 pip install cuda-cccl
 ```
 
-**Install from source** (for development):
+From source:
+
 ```bash
 git clone https://github.com/NVIDIA/cccl.git
 cd cccl/python/cuda_cccl
-pip install -e .[test]  # Development mode with test dependencies
+pip install -e .[test]
```

Comment:
Today, this is `[test-cu12]` or `[test-cu13]` depending on what CUDA version is available.